### DISCUSSION TITLE
How to run an instance of a parameterized fixture only against a certain test module.

### DISCUSSION THREAD

**Discussion by suhaibasaeed** [pytest-dev/pytest#10653] [Q&A] [answered]
After spending quite some time trawling through documentation about parameterized fixtures, I was unable to figure out if it's possible to use separate instances of a parameterized fixture only with a certain test module.

My parametrized fixture inside my conftest.py below hits an API and gets a dictionary of devices to test against. I have separate test modules for different device types (rr, cr, bdr and pe). 

Is it possible to use an instance of my fixture i.e. request.params=rr only for the rr test module? Or would I have to copy and paste the fixture into each test module instead and then change the one parameter?

```
@pytest.fixture(params=["rr", "cr", "bdr", "pe"], scope="session", autouse=True)
def pytestnr(request):
    # Initalise nornir
    pytestnr = InitNornir(
        inventory={
            "plugin": "NetBoxInventory2",
            "options": {
                "nb_url": my_secrets["netbox_url"],
                "nb_token": my_secrets["netbox_token"],
                "ssl_verify": False,
                "use_platform_napalm_driver": True,
            },
        },
        logging={"enabled": False},
    )
    pytestnr = pytestnr.filter(F(name__contains=request.param))
```

Thanks in advance!

**Comment by RonnyPfannschmidt** [comment, marked as answer]
Parametrize happens before fixtures, so no

**Reply by suhaibasaeed** [reply]
Thanks for the prompt answer!
